Feature/boss hitbox

Toine2106 opened the pull request:
new hitboxes and animations for the boss

Toine2106 committed:
Setup rotate hitbox not finished

Toine2106 committed:
Some changes and additions to th hitboxes

Toine2106 committed:
Merge branch 'dev' of https://github.com/smooth-soup-studios/oh-my-cult into feature/boss-hitbox

Toine2106 committed:
Hitboxes boss fixed

Toine2106 committed:
Hitbox turn çmmmmmhm

Toine2106 committed:
Boss hitboxes rotate now flash needs to be fixed

Toine2106 committed:
New boss animation
fixed red circle flash before attack

Justwouter committed:
Merge branch 'dev' into feature/boss-hitbox

Justwouter commented:
The oval shape for the side slam looks kinda weird. The side slam is also triggered while the player is not in its range and the animation offset makes the boss do a weird jump.
Boss hitbox is a bit irregular with the top side being imo a bit to big and the bottom side allowing you to float into his mouth.
The boss can and will move trough the arena boundaries.
When standing lower than the boss, it will do a walk for .1 seconds before attacking while the player is already in range.
Can softlock the boss by standing outside the circle collider between the side/up-down collider

AI is a lot better tough, the walking towards the player was really necessary.

# Art stuff
The boss outline turns purple for a second when walking this is probably an issue with the sprites.
Boss's horns are diffrently sized in side and front/back anims
Slam up just looks weird maybe replace with a punch or smth?

Justwouter committed:
Merge branch 'dev' into feature/boss-hitbox

Justwouter committed:
Resized boss hitboxes,
Added church sprites to atlas

Toine2106 committed:
Changed some hitboxes from boss to right size and without blind spots
changed walking range and speed from boss
added coroutine so the boss doesn't move on scene load and waits behind the altar

Justwouter committed:
Merge branch 'dev' into feature/boss-hitbox

Justwouter committed:
Merge branch 'dev' into feature/boss-hitbox
